### PR TITLE
pysnmp 7.1.15

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-6089](https://anaconda.atlassian.net/browse/PKG-6089)
- [Upstream repository](https://github.com/etingof/pysnmp)

### Explanation of changes:

- Update version
- Remove run_test.py
- Update test import and add pip check
- linter fixes

[PKG-6089]: https://anaconda.atlassian.net/browse/PKG-6089?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -80,8 +66,7 @@ about:
     features fully-functional SNMP engine capable to act in Agent/Manager/Proxy
     roles, talking SNMP v1/v2c/v3 protocol versions over IPv4/IPv6 and other
     network transports.
-  doc_url: http://pysnmp.sourceforge.net/docs/tutorial.html
-  doc_source_url: https://github.com/etingof/pysnmp/tree/master/docs/source/docs
+  doc_url: https://pysnmp.sourceforge.net/docs/tutorial.html
   dev_url: https://github.com/etingof/pysnmp
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  dev_url: https://github.com/lextudio/pysnmp
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -80,8 +66,7 @@ about:
     features fully-functional SNMP engine capable to act in Agent/Manager/Proxy
     roles, talking SNMP v1/v2c/v3 protocol versions over IPv4/IPv6 and other
     network transports.
-  doc_url: http://pysnmp.sourceforge.net/docs/tutorial.html
-  doc_source_url: https://github.com/etingof/pysnmp/tree/master/docs/source/docs
+  doc_url: https://pysnmp.sourceforge.net/docs/tutorial.html
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  doc_url: https://docs.lextudio.com/pysnmp/
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -68,9 +52,11 @@ test:
     - pysnmp.smi
     - pysnmp.smi.mibs
     - pysnmp.smi.mibs.instances
+  commands:
+    - pip check
 
 about:
-  home: http://snmplabs.com/pysnmp
+  home: https://snmplabs.com/pysnmp
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  home: https://www.pysnmp.com/pysnmp/
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,55 +1,39 @@
...
+    - poetry-core >=1.0.0
     - pip
   run:
     - python
-    - pyasn1 >=0.2.3
-    - pysmi
-    - pycryptodomex
-    - twisted
+    - pyasn1 >=0.4.8,!=0.5.0
+    - pysmi >=1.3.0,<2.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:23]: pysmi is not listed as a required dependency, but it was set before, and is also set on the conda-forge recipe. So ok to keep.

**Review by cbouss — COMMENTED** [review]
Same as pysmi, the hosting site has changed.

File: recipe/meta.yaml (hunk 4 above)
Comment (SC426) [reply recipe/meta.yaml:23]: Ok thanks!

**Review by ELundby45 — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**